src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -32,14 +32,12 @@ private void addBefore(T value, Node<T> before) {
     }
 
     private Node<T> getNodeByIndex(int index) {
-        Node<T> temp;
-        if (index <= size && index >= 0) {
+        Node<T> temp = null;
+        if (indexExists(index)) {
             temp = first;
             for (int i = 0; i < index; i++) {
                 temp = temp.next;
             }
```

Comment:
you don't need this if block at all, if your index is incorrect the exception will be thrown, so let's do it like this:
indexExists(index);
Node<T> temp = first;
...